### ISSUE TITLE
Question: Modules structure

### ISSUE THREAD

**Issue by zverbeta** [edcarroll/ng2-semantic-ui#176]
Official docs about module/components structure https://angular.io/guide/styleguide#overall-structural-guidelines

Do you plan to follow it?

**Comment by edcarroll** [comment]
This project has gone through a number of quite drastic restructures in it's history, in order to make things easier to understand.

The layout I'm using is:

* Module Type (Semantic UI types, e.g. `collections`, `modules`, ...)
  * Module Name
     * `index.ts` - exports **everything** in the module
     * `public.ts` - exports only **public facing** module exports
     * `${moduleName}.module.ts` - NgModule definition
     * Export Type (Following Angular names, e.g. `components`, `directives`, `services`, ...)
         * `...${fileName}.ts` - Files that export one or more things of the type specified by the outer folder

I've used this layout for the following reasons:

* Outer folders clearly use Semantic UI categories so people can quickly find what they need
* Module folders are split into the name of what they are
* Inside a module folder are barrel files (`public.ts` and `index.ts` that make it very easy to import modules into eachother without any crazy file rename fragility)
* Files inside modules are categorized into sensible Angular based categories again to be able to quickly find what one needs

So to summarise, I believe I am following the styleguide to an extent, but find that it is not as a whole suitable for this library.

Let me know if you have any more questions, I hope I've answered you sufficiently 😄
